iceoryx_binding_c/source/c_client.cpp

```diff
@@ -77,19 +77,20 @@ iox_client_t iox_client_init(iox_client_storage_t* self,
         clientOptions.serverTooSlowPolicy = c2cpp::consumerTooSlowPolicy(options->serverTooSlowPolicy);
     }
 
-    new (self) UntypedClient(ServiceDescription{IdString_t(TruncateToCapacity, service),
-                                                IdString_t(TruncateToCapacity, instance),
-                                                IdString_t(TruncateToCapacity, event)},
-                             clientOptions);
+    auto* me = new UntypedClient(ServiceDescription{IdString_t(TruncateToCapacity, service),
+                                                    IdString_t(TruncateToCapacity, instance),
+                                                    IdString_t(TruncateToCapacity, event)},
+                                 clientOptions);
 
-    return reinterpret_cast<iox_client_t>(self);
+    self->do_not_touch_me[0] = reinterpret_cast<uint64_t>(me);
```

Comment:
Why even write the pointer to the storage? I thought the storage is completely discarded. Maybe I find out during further review.

Maybe to have an additional way to get the pointer ... but the user should just use the returned pointer and store it or use it.

Same comment applies for all similar iceoryx storage structs.

Reply:
This is also just in case we overlooked something for CycloneDDS and it becomes necessary to access the actual object from the storage